tests/test_default_values.py

```diff
@@ -0,0 +1,214 @@
+"""
+
+import unittest
+import datetime
+
+import configsuite
+from configsuite import MetaKeys as MK
+from configsuite import types
+from .data import car
+
+
+class TestDefaultValues(unittest.TestCase):
+    def test_return_empty_containers(self):
+        raw_config = car.build_all_default_config()
+        car_schema = car.build_schema()
+        config_suite = configsuite.ConfigSuite(raw_config, car_schema)
+
+        self.assertTrue(config_suite.valid)
+
+        # Unspecified containers are defaulted to empty container of themselves
+        self.assertEqual((), config_suite.snapshot.owner)
+        self.assertEqual((), config_suite.snapshot.incidents)
+
+        self.assertEqual(None, config_suite.snapshot.color.chassis)
+        self.assertEqual(None, config_suite.snapshot.color.interior)
+
+    def test_default_values(self):
+        raw_config = car.build_default_config()
+        car_schema = car.build_schema()
+        config_suite = configsuite.ConfigSuite(raw_config, car_schema)
+        self.assertTrue(config_suite.valid)
+
+        # Named dict top level
+        self.assertEqual(15, config_suite.snapshot.tire.dimension)
+        self.assertEqual("green", config_suite.snapshot.tire.rim)
+
+        # Named dict within dict
+        owner_data = (("Johan", "Svalbard"), ("Svein", "Earth"))
+        self.assertEqual(len(owner_data), len(config_suite.snapshot.owner))
+        for (expected_name, expected_location), config_owner in zip(
+            owner_data, sorted(config_suite.snapshot.owner)
+        ):
+            self.assertEqual(expected_name, config_owner.value.name)
+            self.assertEqual(expected_location, config_owner.value.location)
+
+        # Named dict within list
+        incidents_data = (("whereabouts", 0), ("somewhere else", 1))
+        self.assertEqual(len(incidents_data), len(config_suite.snapshot.incidents))
+        for (expected_location, expected_casualties), config_owner in zip(
+            incidents_data, sorted(config_suite.snapshot.incidents)
+        ):
+            self.assertEqual(expected_location, config_owner.location)
+            self.assertEqual(expected_casualties, config_owner.casualties)
+
+    def test_element_transformations_applied(self):
+        raw_config = car.build_all_default_config()
+        car_schema = car.build_schema()
+        tire_schema = car_schema[MK.Content]["tire"]
+        tire_schema[MK.Content]["dimension"][MK.Transformation] = car.inch_to_cm
+        config_suite = configsuite.ConfigSuite(raw_config, car_schema)
+
+        self.assertTrue(config_suite.valid)
+        self.assertAlmostEqual( 17 * 2.54, config_suite.snapshot.tire.dimension)
+
+    def test_context_transformation_applied(self):
+        raw_config = car.build_all_default_config()
+        car_schema = car.build_schema()
+        tire_schema = car_schema[MK.Content]["tire"]
+        tire_schema[MK.Content]["dimension"][MK.ContextTransformation] = car.inch_to_cm
```

Comment:
This will fail due to the context being passed as an argument. You need to wrap the `inch_to_cm` in another transformator taking the context, but ignoring it. I think I prefer the explicit second version over making `inch_to_cm` take `*args` to facilitate future refactoring. Something about explicit over implicit and those things.... ;)

Reply:
Right, true.. I'll think of something else, I agree with being explicit..

Reply:
I think something like this should do the trick:

```python
@transformation_msg("Transforms from inches to cm with an ignored context")
def inch_to_cm_context(l, _):
    return inch_to_cm(l)
```

No need to change the rest of the test, just swap `inch_to_cm` with `inch_to_cm_context`...

Reply:
Yeh, I know..

But I often look at the tests to see how things are supposed to function, and the more tests that actually reflect that the better.. That was the philosophy behind why I changed it to behave as intended.

There's a balance though of course.. Tests should be made to test only one thing at the time..

Reply:
But to be honest, the test became cluttered and wasn't really that helpful :P